File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public int[] solution(int n) {
        List<Integer> ret = new ArrayList<>();
        for(int i = 1; i * i <= n; i++){
            if(n % i == 0) {
                ret.add(i);
                if(i * i == n) continue;
                ret.add(n / i);
            }
        }
        return ret.stream().sorted().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.py
def solution(n):
    if n == 1: return [1]
    answer = set()
    for i in range(1, n+1):
        if i * i > n: break
        if n % i == 0:
            answer.add(i)
            answer.add(n//i)
    return sorted(list(answer))